### PR TITLE
Added a method to refresh the robot

### PR DESCRIPTION
**Pull request by geoffreychiou**
@AustinDeric  please review.

### CODE REVIEW

File: moveit_simple/src/robot.cpp [hunk 1]
```diff
@@ -98,6 +80,33 @@ Robot::Robot(const ros::NodeHandle &nh, const std::string &robot_description,
 
 Robot::~Robot() { }
 
+void Robot::refreshRobot()
+{
+  ROS_INFO_STREAM("Loading MoveIt objects based on, robot description: " << robot_description_
+    << ", group name: " << planning_group_);
+
+  std::lock_guard<std::recursive_mutex> guard(m_);
```
Comment (AustinDeric) [inline moveit_simple/src/robot.cpp:88]: :1st_place_medal:

File: moveit_simple/src/robot.cpp [hunk 2]
```diff
@@ -98,6 +80,33 @@ Robot::Robot(const ros::NodeHandle &nh, const std::string &robot_description,
...
+
+  ROS_INFO_STREAM("Calculating all positions assuming, root: " << robot_model_ptr_->getRootLinkName()
+    << ", and tool: " << robot_model_ptr_->getLinkModelNames());
+  ROS_INFO_STREAM("MoveIt object loaded");
+
+  virtual_visual_tools_.reset(new moveit_visual_tools::MoveItVisualTools(robot_model_ptr_->getRootLinkName(),
+    nh_.getNamespace() + "/rviz_visual_tools", robot_model_ptr_));
+  virtual_visual_tools_->loadRobotStatePub(nh_.getNamespace() + "/display_robot_state");
+  virtual_visual_tools_->publishRobotState(virtual_robot_state_, rviz_visual_tools::PURPLE);
+  virtual_visual_tools_->trigger();
```
Comment (AustinDeric) [inline moveit_simple/src/robot.cpp]: :)

Comment (AustinDeric) [inline moveit_simple/src/robot.cpp:107]: :+1:

**Review by AustinDeric — APPROVED** [review]
Good stuff! light changes but impactful!

**Comment by AustinDeric** [comment]
I'll make an issue to make a test for this.

**Comment by geoffreychiou** [comment]
@AustinDeric, @130s. This MR is no longer needed, do you guys think it is worth reverting or leaving the changes in?

**Comment by 130s** [comment]
> This MR is no longer needed, do you guys think it is worth reverting or leaving the changes in?

I'd say let's keep anything that we can't think of cases where it causes something bad.

**Pull request merged**